src/Shared/HttpSys/RequestProcessing/NativeRequestContext.cs

```diff
@@ -478,5 +479,42 @@ private uint GetChunksHelper(ref int dataChunkIndex, ref uint dataChunkOffset, b
             }
             return dataRead;
         }
+
+        internal IReadOnlyCollection<KeyValuePair<int, ReadOnlyMemory<byte>>> GetRequestInfo()
+        {
+            if (_permanentlyPinned)
+            {
+                return GetRequestInfo((IntPtr)_nativeRequest, (HttpApiTypes.HTTP_REQUEST_V2*)_nativeRequest);
+            }
+            else
+            {
+                fixed (byte* pMemoryBlob = _backingBuffer)
+                {
+                    var request = (HttpApiTypes.HTTP_REQUEST_V2*)(pMemoryBlob + _bufferAlignment);
+                    return GetRequestInfo(_originalBufferAddress, request);
+                }
+            }
+        }
+
+        private IReadOnlyCollection<KeyValuePair<int, ReadOnlyMemory<byte>>> GetRequestInfo(IntPtr baseAddress, HttpApiTypes.HTTP_REQUEST_V2* nativeRequest)
+        {
+            if (nativeRequest->RequestInfoCount == 0)
```

Comment:
Nit: probably faster to save nativeRequest->RequestInfoCount in a local

Reply:
Cleaner, but not faster.

Reply:
yeah cause pointers, not properties.